code/modules/mob/living/simple_animal/friendly/drone/_drone.dm

```diff
@@ -121,6 +123,24 @@
 	"<span class='notice'>     - Interacting with non-living beings (dragging bodies, looting bodies, etc.)</span>\n"+\
 	"<span class='warning'>These rules are at admin discretion and will be heavily enforced.</span>\n"+\
 	"<span class='warning'><u>If you do not have the regular drone laws, follow your laws to the best of your ability.</u></span>"
+	/// blacklisted drone areas, direct
+	var/list/drone_area_blacklist_flat = list(/area/engineering/atmos, /area/engineering/atmospherics_engine, /area/engineering/supermatter)
+	/// blacklisted drone areas, recursive/includes descendants
+	var/list/drone_area_blacklist_recursive
+	/// whitelisted drone machines, direct
+	var/list/drone_machinery_whitelist_flat
+	/// whitelisted drone machines, recursive/includes descendants
+	var/list/drone_machinery_whitelist_recursive = list(	/obj/machinery/atmospherics, /obj/machinery/autolathe, /obj/machinery/cell_charger, /obj/machinery/disposal, \
+															/obj/machinery/drone_dispenser, /obj/machinery/light, /obj/machinery/pipedispenser, /obj/machinery/recharger, \
+															/obj/machinery/rnd/production)
+	/// whitelisted drone items, direct
+	var/list/drone_item_whitelist_flat = list(	/obj/item/crowbar/drone, /obj/item/screwdriver/drone, /obj/item/wrench/drone, \
+												/obj/item/weldingtool/drone, /obj/item/wirecutters/drone )
+	/// whitelisted drone items, recursive/includes descendants
+	var/list/drone_item_whitelist_recursive = list(	/obj/item/airlock_painter, /obj/item/conveyor_switch_construct, /obj/item/electronics, /obj/item/light, \
+													/obj/item/pipe_meter, /obj/item/stack/cable_coil, /obj/item/stack/circuit_stack, /obj/item/stack/conveyor, \
+													/obj/item/stack/pipe_cleaner_coil, /obj/item/stack/rods, /obj/item/stack/sheet, /obj/item/stack/tile, \
+													/obj/item/stock_parts, /obj/item/toner, /obj/item/clothing/head, /obj/item/clothing/mask )
```

Comment:
will drones no longer be able to wear buckets (without somehow getting a human to help them, anyway)?

Reply:
should rcds and rpds be added to this list? what about airlock electronics?

Reply:
Airlock electronics are an `/obj/item/electronics`

RCDs and RPDs intentionally missing: I don't want drones to use them. As I said in the PR body, I want them to be weaker.

I could add a bucket, if you deem it necessary to wear a bucket :wink:

Reply:
well, the problem with adding a bucket is that a bucket can be used to hold reagents.

Reply:
Sorry bucket, you will be missed